situp-plugins/apmtracesource/src/main/java/com/amazon/situp/plugins/processor/oteltrace/RawTrace.java

```diff
@@ -0,0 +1,145 @@
+        private String serviceName;
+        private Map<String, String> resourceAttributes;
+
+        public RawTraceBuilder() {
+        }
+
+        public RawTraceBuilder setTraceID(String traceID) {
+            checkNotNull(traceID, "traceID cannot be null");
+            this.traceID = traceID;
+            return this;
+        }
+
+        public RawTraceBuilder setSpanID(String spanID) {
+            checkNotNull(spanID, "spanID cannot be null");
+            this.spanID = spanID;
+            return this;
+        }
+
+        public RawTraceBuilder setParentSpanID(String parentSpanID) {
+            this.parentSpanID = parentSpanID;
+            return this;
+        }
+
+        public RawTraceBuilder setName(String name) {
+            this.name = name;
+            return this;
+        }
+
+        public RawTraceBuilder setStartTime(String startTime) {
+            this.startTime = startTime;
+            return this;
+        }
+
+        public RawTraceBuilder setEndTime(String endTime) {
+            this.endTime = endTime;
+            return this;
+        }
+
+        public RawTraceBuilder setInstrumentationName(String instrumentationName) {
+            this.instrumentationName = instrumentationName;
+            return this;
+        }
+
+        public RawTraceBuilder setInstrumentationVersion(String instrumentationVersion) {
+            this.instrumentationVersion = instrumentationVersion;
+            return this;
+        }
+
+        public RawTraceBuilder setServiceName(String serviceName) {
+            this.serviceName = serviceName;
+            return this;
+        }
+
+        public RawTraceBuilder setResourceAttributes(Map<String, String> resourceAttributes) {
+            this.resourceAttributes = resourceAttributes;
+            return this;
+        }
+
+        public RawTrace build() {
+            return new RawTrace(this);
+        }
+
+    }
+
+    private static String convertStringNanosToISO8601(final String stringNanos) {
+        final BigDecimal nanos = new BigDecimal(stringNanos);
+        final long epochSeconds = nanos.divide(MILLIS_TO_NANOS.multiply(SEC_TO_MILLIS), RoundingMode.DOWN).longValue();
+        final int nanoAdj = nanos.remainder(MILLIS_TO_NANOS.multiply(SEC_TO_MILLIS)).intValue();
+        return Instant.ofEpochSecond(epochSeconds, nanoAdj).toString();
+    }
+
+    public static RawTrace buildFromProto(Resource resource, Span spans, InstrumentationLibrary instrumentationLibrary) {
+        final String SERVICE_NAME = "service.name";
+
+        return new RawTraceBuilder()
+                .setTraceID(Hex.encodeHexString(spans.getTraceId().toByteArray()))
+                .setSpanID(Hex.encodeHexString(spans.getSpanId().toByteArray()))
+                .setParentSpanID(Hex.encodeHexString(spans.getParentSpanId().toByteArray()))
+                .setName(spans.getName())
+                .setStartTime(convertStringNanosToISO8601(String.valueOf(spans.getStartTimeUnixNano())))
```

Comment:
can spans.getStartTimeUnixNano be 0 or null [think it cannot be null as it is long] ?
If its 0, ur conversion would return `1970-01-01T00:00:00` - is this fine ? or should we be doing something smart here ?

Reply:
@kowshikn its a nit but how do u think we should handle this - we can leave it and be 1970 or we can just put null and let sink (es) handle it

Reply:
I think zero mapped to Unix epoch start tome is reasonable. If this is an issue we can address it later.